core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -184,16 +186,21 @@ class RequestSendThread(val controllerId: Int,
               backoff()
             }
             else {
-              val requestHeader = apiVersion.fold(networkClient.nextRequestHeader(apiKey))(networkClient.nextRequestHeader(apiKey, _))
-              val clientRequest = new ClientRequest(brokerNode.idString, time.milliseconds(), true, requestHeader, request, null)
+              val version = apiVersion match  {
+                case None => ProtoUtils.latestVersion(apiKey.id)
+                case Some(v) => v
+              }
```

Comment:
These 3 lines can simply be `val version = apiVersion.getOrElse(ProtoUtils.latestVersion(apiKey,id))`.